### PR TITLE
Adds a basic go to definition usage tutorial

### PR DESCRIPTION
**Pull request by gradha**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -41,6 +43,67 @@ au User asyncomplete_setup call asyncomplete#register_source({
...
+when isMainModule:
+  main()
+```
+
+Put your cursor on the first `testMe()` call inside the `main()` proc, and
+while still in normal mode, type the default binding to jump to the definition
+of the `testMe()` proc, which is `gd`. If the plugin is correctly installed and
+nimsuggest is working, your cursor should jump to the sixth column of the third
+line, where the proc is defined. Type `Ctrl+o` to jump back, and repeat the
+same on the line below: since Nim syntax is case insensitive, the go to
```
Comment (alaviss) [inline readme.md]:
```suggestion
same on the line below: since Nim syntax is style insensitive, the go to
```

File: readme.md [hunk 2]
```diff
@@ -41,6 +43,67 @@ au User asyncomplete_setup call asyncomplete#register_source({
...
+definition command will work the same on the `test_me()` call.
+
+Now position your cursor on the `toLower` call at the end of the fourth line
+and repeat the go to definition command. Nimsuggest works across source files
+and into Nim's own standard library, so you should end up looking at the
+`toLower*(s: string)` proc, which at the moment of writing this is located on
+[line 1546 of the lib/pure/unicode.nim
+file](https://github.com/nim-lang/Nim/blob/aa072b952555804e516861d9ed62846e341f7938/lib/pure/unicode.nim#L1546).
```
Comment (alaviss) [inline readme.md]: I would prefer to have this annotated with `[1]` and have the link referenced at the end of the file.

File: readme.md [hunk 3]
```diff
@@ -41,6 +43,67 @@ au User asyncomplete_setup call asyncomplete#register_source({
...
+You can even move your cursor a few lines below to the `convertRune(s,
+toLower)` and nimsuggest will keep working. Type `Ctrl+o` (several times, if
+needed) to jump back to your source file.
+
+If you are working on a file, chances are it will be _dirty_ and not saved to
+disk. In these situations, the go to definition command only works for your own
+source. If you start any modification of the test file and try to go to the
+definition of `toLower`, you will be greeted with the error `E37: No write
```
Comment (alaviss) [inline readme.md]: I think the error part deserves it's own line, like this:

```markdown
you will be greeted with the error:

    E37: No write since last change (add ! to override)

This is because Neovim ...
```

File: readme.md [hunk 4]
```diff
@@ -41,6 +43,67 @@ au User asyncomplete_setup call asyncomplete#register_source({
...
+since last change (add ! to override)` because Neovim does not want to lose
+your changes switching the current buffer. You can avoid this in two ways: save
```
Comment (alaviss) [inline readme.md]: For this part, I believe it's best to present it like this:

```markdown
This can be avoided in two ways:
- Save the file beforehand
- Use the alternative `gD` mapping, which will ...
```

I found long paragraphs to be rather hard to follow, and/or hard to locate a specific part in the case I forgot how to use something, so I think some structuring should be applied to split this into easier to locate chunks.

File: readme.md [hunk 5]
```diff
@@ -41,6 +43,67 @@ au User asyncomplete_setup call asyncomplete#register_source({
...
+(upper case `D`). This alternative mapping will split the current window and go
+to the definition in the new window. Typing `Ctrl+o` now would bring you back
+to your buffer and keep the split window, so you might as well close the window
+directly typing `:q`.
+
+If you look at the default plug-in key mappings at the end of
+[ftplugin/nim.vim](ftplugin/nim.vim), you will see the
+[gd](ftplugin/nim.vim#L63) and [gD](ftplugin/nim.vim#L64) mappings. If you
+prefer splitting your window vertically, note that there is an orphan
+[NimGoToDefVSplit](ftplugin/nim.vim#L61). You can use it adding the following
```
Comment (alaviss) [inline readme.md]:
```suggestion
[NimGoToDefVSplit](ftplugin/nim.vim#L61). You can use it by adding the following
```

However, I think that we should empathise that these `<Plug>` directives are for the user to define their own mapping, like this:

````markdown
This plugin provides several convience mapping which could be used to
create your preferred custom mapping:

- `<Plug>NimGoToDefBuf`: Go to definition using the current buffer
- `<Plug>NimGoToDefSplit`: Open the definition in a horizontal split
- `<Plug>NimGoToDefVSplit`: Open the definition in a vertical split

For example, if you prefer having a mapping to split the window vertically,
add this to your `init.vim` file:

```vim
nmap gV <Plug>NimGoToDefVSplit
```

Which will map `gV` to the desired function.
````

File: readme.md [hunk 6]
```diff
@@ -41,6 +43,67 @@ au User asyncomplete_setup call asyncomplete#register_source({
...
+line to your Neovim `init.vim` file:
+
+```
+nmap gV <Plug>NimGoToDefVSplit
+```
+
+
```
Comment (alaviss) [inline readme.md]: Of course, we should also let the user know that they can just replace the `gD` with `VSplit` if it's preferred :)

File: readme.md [hunk 7]
```diff
@@ -41,6 +43,67 @@ au User asyncomplete_setup call asyncomplete#register_source({
...
+your changes switching the current buffer. You can avoid this in two ways: save
+the file before going to the definition, or typing the alternative mapping `gD`
+(upper case `D`). This alternative mapping will split the current window and go
+to the definition in the new window. Typing `Ctrl+o` now would bring you back
+to your buffer and keep the split window, so you might as well close the window
+directly typing `:q`.
+
+If you look at the default plug-in key mappings at the end of
+[ftplugin/nim.vim](ftplugin/nim.vim), you will see the
+[gd](ftplugin/nim.vim#L63) and [gD](ftplugin/nim.vim#L64) mappings. If you
```
Comment (alaviss) [inline readme.md]: I would prefer that we don't point our users to the source, as the documentations are supposed to help the user avoid having to read the sources.

**Comment by gradha** [comment]
Sorry about that.
